pkgs/by-name/co/convmv/package.nix

```
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  perl,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "convmv";
+  version = "2.05";
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  src = fetchzip {
+    url = "https://www.j3e.de/linux/convmv/convmv-${finalAttrs.version}.tar.gz";
+    hash = "sha256-ts9xAPRGUoS0XBRTmpb+BlGW1hmGyUs+rQLyUEgiZ54=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [ perl ];
+
+  buildInputs = [ perl ];
+
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "MANDIR=${placeholder "man"}/share/man"
+  ];
+
+  checkTarget = "test";
+
+  # testsuite.tar contains filenames that aren't valid UTF-8. Extraction of
+  # testsuite.tar will fail as APFS enforces that filenames are valid UTF-8.
+  doCheck = !stdenv.isDarwin;
+
+  prePatch =
+    lib.optionalString finalAttrs.doCheck ''
+      tar -xf testsuite.tar
+    ''
+    + ''
+      patchShebangs --host .
+    '';
```

I find the order a bit unintuitive. Do files unpacked from `testsuite.tar` also need the shebangs patched? If not I'd find the following more intuitive:

```suggestion
  prePatch = ''
      patchShebangs --host .
    '' + lib.optionalString finalAttrs.doCheck ''
      tar -xf testsuite.tar
    '';
```

---

Yes, `testsuite.tar` contains `parsable_tester.pl` with the following shebang:

```perl
#!/usr/bin/perl
# this is just a simple test script for use in the convmv test suite and not
# made for general use
```

It needs to be patched, otherwise the tests fail:

```
Ready! I converted 19 files in 0 seconds.
parsable-test
./dotests.sh: line 153: ./parsable_tester.pl: not found
Unable to flush stdout: Broken pipe
parsable-test failed.
make: *** [Makefile:25: test] Error 1
```

---

Thanks for confirming, @al3xtjames 🙂 👍